tools/osbuild-mpp

```diff
@@ -840,10 +846,24 @@ class PartitionTable:
         command = f"label: {self.label}\nlabel-id: {self.uuid}"
         for partition in self.partitions:
             fields = []
-            for field in ["start", "size", "type", "name", "uuid"]:
+            for field in ["start", "size", "type", "name", "uuid", "attrs"]:
                 value = getattr(partition, field)
-                if value:
-                    fields += [f'{field}="{value}"']
+                if not value:
+                    continue
+                if field == "attrs":
+                    resv = {
+                        0: "RequiredPartition",
+                        1: "NoBlockIOProtocol",
+                        2: "LegacyBIOSBootable"
+                    }
+                    attrs = []
+                    for bit in value:
+                        if bit in resv:
+                            attrs += [f'{resv[bit]}']
+                        elif 48 <= bit <= 63:
+                            attrs += [f'{bit}']
```

Comment:
Minor codestyle nitpick: This could be simplified to `attrs.append(resv.get(bit, str(bit)))` (or it could be done in two steps for a bit of clarity, one for `get()` and one for `append()`)

Reply:
I made the change before I realized this will now silently fail if an invalid bit is passed, setting none of the bits if at least one is invalid (any bit that is not `[0-2]` or `[48-63]`). Turns out `sfdisk` returns `0`, only when reading a script, if setting the attribute bits fail.

Reply:
I am the one that should apologize, I pushed the change and did not test it correctly :).
I reverted that part to ignore bits that `sfdisk` would not set.